includes/class-wc-payment-discounts-add-discount.php

```diff
@@ -47,7 +47,7 @@ protected function calculate_discount( $amount, $subtotal ) {
 			$amount = ( $subtotal / 100 ) * str_replace( '%', '', $amount );
 		}
 
-		return $amount;
+		return apply_filters( 'wcpd-amount', $amount ); $amount;
```

Comment:
actually... it was correct. You see, the red line has a minus in front - meaning it was removed. The green line has a plus-sign in front - meaning it was added. 

Tested - works.